### PR TITLE
Add inscription charms

### PR DESCRIPTION
**Pull request by casey**
Charms are cute emoji that appear on inscription pages. Charms include rarity charms, for inscriptions on rare sats, cursed inscriptions, inscriptions on block 9 sats, lost sats, unbound inscriptions, and reinscriptions.

### CODE REVIEW

**Comment by devords** [comment]
If we're able to show the sequence number for the specific kind of curse, I think that'll be really really valuable. Sort of like an unstable inscription # for specific kind of curses.

Pretty please 🙏

File: src/charm.rs [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    charms & self.flag() != 0
+  }
+
+  pub(crate) fn icon(self) -> &'static str {
+    match self {
+      Charm::Cursed => "👹",
+      Charm::Epic => "🪻",
+      Charm::Legendary => "🌝",
+      Charm::Lost => "🤔",
+      Charm::Nineball => "🌅",
```
Comment (raphjaph) [inline src/charm.rs:45]:
```suggestion
      Charm::Nineball => "❾",
```

**Review by raphjaph — APPROVED** [review]

**Comment by raphjaph** [comment]
🪄

**Comment by casey** [comment]
> If we're able to show the sequence number for the specific kind of curse, I think that'll be really really valuable. Sort of like an unstable inscription # for specific kind of curses.
> 
> Pretty please 🙏

Showing sequence is a little bit controversial. I'm in favor of doing it, but it should probably be discussed beforehand. Feel free to open an issue!

**Pull request merged**